### ISSUE TITLE
Ensure macros in McBopomofo macOS and fcitx5-mcbopomofo are in sync

### ISSUE THREAD

**Issue by lukhnos** [openvanilla/fcitx5-mcbopomofo#100]
Two concerns from maintenance perspective:
- We want to make sure that macros introduced on one platform are supported in another; if not always feasible, McBopomofoLM should be modified such that unsupported macros are removed from the candidtae list
- Making changes to `data.txt` in fcitx5-mcbopomofo are not sustainable, since any sync with the upstream could cause changes to be overwritten

For now I can see some short-term solutions:
- Require that changes to `data.txt` are also reflected in the upstream data source
- Have a unit test (maybe just a Python script) that checks macros extracted from fcitx5-mcbopomofo source satisfy those in `data.txt`

**Comment by xatier** [comment]
Perhaps a simple shell one-liner can do?

```
diff <(grep 'MACRO@' src/InputMacro.cpp | perl -nE 'say $1 if /.*\(\"(\S+?)\".*/' | sort | uniq) <(grep 'MACRO@' data/data.txt | awk '{print $2}' | sort | uniq)
```

The issue here is that simple grep may break if we construct the code in some other way later. The more robust way to test this is to add some methods that expose the internal `unordered_map macros_` and build some gtests around it?

**Comment by zonble** [comment]
I guess it is time to move the data builder from the repo of the Mac version to another project, and let all of the versions of McBopomofo on various platforms to depend on the data repo as a git submodule, so each version can get the latest data, or data in a specific commit.

On the other hand, since all of the macros have a unified prefix, McBopomofoLM should just identify the candidates without conversion and remove them.

**Comment by lukhnos** [comment]
In the end this is addressed via #133. Like @zonble said:

> On the other hand, since all of the macros have a unified prefix, McBopomofoLM should just identify the candidates without conversion and remove them.

That's the approach we've taken. It may have an additional upside in that now the macOS version and fcitx5 version can support differernt macros without requiring developers to keep everything in sync.
